Server/ReasnAPI/ReasnAPI/Services/ParticipantService.cs

```diff
@@ -1,16 +1,94 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
 
 namespace ReasnAPI.Services {
     public class ParticipantService (ReasnContext context) {
         private readonly ReasnContext _context = context;
 
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+        public ParticipantDto? CreateParticipant(ParticipantDto? participantDto) {
+            if (participantDto is null)
+                return null;
+
+            // check if participant already exists (same event and user)
+            var participantDb = _context.Participants.FirstOrDefault(r => r.Event.Id == participantDto.EventId && r.User.Id == participantDto.UserId);
+
+            if (participantDb is not null)
+                return null;
+
+            var userDb = _context.Users.FirstOrDefault(r => r.Id == participantDto.UserId);
+            var eventDb = _context.Events.FirstOrDefault(r => r.Id == participantDto.EventId);
+            var statusDb = _context.Statuses.FirstOrDefault(r => r.Id == participantDto.StatusId);
```

Comment:
Keep the same naming convention in each service - there are no conflicting names to avoid using names like: user, event and status. But if you want to use such names, try naming it that way: userInDb, eventInDb and statusInDb - current name says that the variable could be the database responsible for user etc. and not the record itself.